[chenyixin0] IP

chenyixin0 opened the pull request:
### TalkingBox: the Task Manager

> Have **you** ever felt: 

- stressed?
- anxious?
- lost?

when looking at your daily to do list? Worry no more! With [this _task manager_ bot](https://www.youtube.com/watch?v=dQw4w9WgXcQ), you will have no problems keeping track of your 

- [ ] todos
- [ ] deadlines
- [ ] key events

features include: 

1. `add` new tasks
2. ~~`delete` tasks~~
3. `find` tasks
4. `mark` tasks as done 👍 or undone 👎

how does it do this? here's a sneak peek:

```
    /**
     * Appends new task to the end of the list
     *
     * @param t  New task to be added
     */
    public void add(Task t) {
        this.taskList.add(t);
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

chenyixin0 committed:
Level 0: Rename, Greet and Exit

chenyixin0 committed:
Level 0: Rename, Greet, Exit

chenyixin0 committed:
remove newline

chenyixin0 committed:
modified line spacing

chenyixin0 committed:
Chatbot echoes user input

chenyixin0 committed:
Added task list function

chenyixin0 committed:
Added task mark and unmark

chenyixin0 committed:
Added task types

chenyixin0 committed:
Added error handling & improved OOP design

chenyixin0 committed:
Added delete function

chenyixin0 committed:
tasklist saved to hard disk on revision

chenyixin0 committed:
tasklist saved to hard disk on revision

chenyixin0 committed:
exit

chenyixin0 committed:
Merge branch 'branch-Level-7'

chenyixin0 committed:
correctly processes date inputs from user

chenyixin0 committed:
Refactored code to extract out closely related code as classes

chenyixin0 committed:
organised classes into packages

chenyixin0 committed:
Merge tag 'A-Gradle'

chenyixin0 committed:
Added JUnit testing

chenyixin0 committed:
Added JUnit testing

YuvBindal reviewed (commented):
Hey your code logic seems sound, I like the way you have structured your code into a good amount of subclasses. For your method header names try to convert them to camelCase. For example EmptyTaskException can be changed to emptyTaskException. Otherwise, solid work keep it up!

chenyixin0 committed:
added JavaDoc to classes/methods

chenyixin0 committed:
Tweak code to comply with coding standard

Make modifications to code to more closely follow Java coding standard

chenyixin0 committed:
Add find task feature

Include command for finding tasks containing a given keyword or key phrase

chenyixin0 committed:
Resolve merge conflict

chenyixin0 committed:
Include small edits

Small changes made to follow coding standard

chenyixin0 committed:
Resolved merge conflict

FangRuoqing commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,30 @@
+
+    public Storage(TaskList taskList, String path) throws FileNotFoundException {
+        this.taskList = taskList;
+        this.path = path;
```
hey, what do you think about renaming path to something like filePath? It'll make it crystal clear what it's all about.

FangRuoqing commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,41 @@
+    LocalDateTime endTime;
+    public Event(String name, String startTime, String endTime) throws DateFormatException {
+        super(name, "E");
+        String[] startString = startTime.split("/");
```
For better clarity, how about renaming variables like startString and endString to something like startDateComponents and endDateComponents? It makes understanding the code much easier!

FangRuoqing reviewed (commented):
Overall, the code adheres to the Java naming conventions, which contributes to its readability and maintainability. By incorporating these changes, the code would become much friendlier and easier to understand for everyone involved. Happy coding!

YuvBindal reviewed (approved):
Hey LGTM. I liked the way you have used enums throughout your code to define your own types. Good work keep it up!

chenyixin0 committed:
Add CheckStyle

Include checStyle to detect coding style violations

chenyixin0 committed:
Merge tag 'A-CheckStyle'

chenyixin0 committed:
Add GUI

Implemented GUI interterface for chatbot

chenyixin0 committed:
Merge branch 'Branch-Level-10'

YuvBindal commented on src/main/java/duke/exception/InvalidIndexException.java:
Good use of inheritance here to modularise your error messages. I like the approach you took it makes it easy to distinguish different error messages as classes rather than methods such that further customisation can be added in the future. The solution is quite scalable.

YuvBindal commented on src/main/java/duke/main/Parser.java:
Perhaps, Parser can be used entirely to handle the string transformation logic?

YuvBindal commented on src/main/java/duke/main/UI.java:
Good adherence to the naming conventions for methods!

YuvBindal reviewed (commented):
Overall, LGTM. The code is solid keep it up!

chenyixin0 committed:
Add notes feature

Currently the chatbot only supports adding and viewing tasks.

The current structure supports adding other kinds of input as well.

Now the chatbot also supports adding and viewing notes by the user.

The notes are stored as a separate list from tasks to minimise possible conflicts.

chenyixin0 committed:
Implement assertions

Adds assertions to check for important assumptions in the code

chenyixin0 committed:
Improve code quality

Previous versions did not fully abide with code quality standards

Changes made to improve logical flow of code

This improves overall readability for others

chenyixin0 committed:
Final changes

chenyixin0 committed:
Add user guide

chenyixin0 committed:
Add user guide

chenyixin0 committed:
Push untracked files

chenyixin0 committed:
Update user guide

chenyixin0 committed:
Improve storage function

Previously the program could not load stored information from hard disk

This has been corrected. Now the program can information the user has previously stored

chenyixin0 committed:
Updated changes
